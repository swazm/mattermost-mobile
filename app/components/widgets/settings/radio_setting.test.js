// Copyright (c) 2015-present SWAZM, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';
import {TouchableOpacity} from 'react-native';
import RadioSetting from './radio_setting.js';
import CheckMark from 'app/components/checkmark';

describe('components/widgets/settings/RadioSetting', () => {
    const theme = Preferences.THEMES.default;
    const options = [
        {text: 'this is engineering', value: 'Engineering'},
        {text: 'this is sales', value: 'Sales'},
        {text: 'this is administration', value: 'Administration'},
    ];

    test('onChange', () => {
        const onChange = jest.fn();
        const wrapper = shallow(
            <RadioSetting
                id='string.id'
                label='some label'
                options={options}
                default={'Administration'}
                onChange={onChange}
                theme={theme}
            />
        );
        wrapper.find(TouchableOpacity).at(1).props().onPress();

        expect(onChange).toHaveBeenCalledTimes(1);
        expect(onChange).toHaveBeenCalledWith('string.id', 'Sales');
    });

    test('should match snapshot when error is present', () => {
        const onChange = jest.fn();
        const wrapper = shallow(
            <RadioSetting
                id='string.id'
                label='some label'
                options={options}
                errorText={'some error message'}
                default={'Administration'}
                onChange={onChange}
                theme={theme}
            />
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('no option should be checked when the value is not in provided options', () => {
        const onChange = jest.fn();
        const wrapper = shallow(
            <RadioSetting
                id='string.id'
                label='some label'
                options={options}
                default={'invalid-option-value'}
                onChange={onChange}
                theme={theme}
            />
        );

        expect(wrapper.find(CheckMark)).toHaveLength(0);
    });
});
