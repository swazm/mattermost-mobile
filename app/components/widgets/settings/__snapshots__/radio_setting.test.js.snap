// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/widgets/settings/RadioSetting should match snapshot when error is present 1`] = `
<View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 10,
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3d3c40",
          "fontSize": 14,
          "marginLeft": 15,
        }
      }
    >
      some label
    </Text>
    <Text
      style={
        Object {
          "color": "#fd5960",
          "fontSize": 14,
        }
      }
    >
       *
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderBottomColor": "rgba(61,60,64,0.1)",
          "borderBottomWidth": 1,
          "borderTopColor": "rgba(61,60,64,0.1)",
          "borderTopWidth": 1,
        },
        null,
      ]
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingHorizontal": 15,
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 45,
            }
          }
        >
          <Text>
            this is engineering
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(61,60,64,0.1)",
            "flex": 1,
            "height": 1,
            "marginLeft": 15,
          }
        }
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingHorizontal": 15,
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 45,
            }
          }
        >
          <Text>
            this is sales
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(61,60,64,0.1)",
            "flex": 1,
            "height": 1,
            "marginLeft": 15,
          }
        }
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingHorizontal": 15,
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 45,
            }
          }
        >
          <Text>
            this is administration
          </Text>
        </View>
        <CheckMark
          color="#2389d7"
          height={12}
          width={12}
        />
      </View>
    </TouchableOpacity>
  </View>
  <View
    style={null}
  >
    <Text
      style={
        Object {
          "color": "#fd5960",
          "fontSize": 12,
          "marginHorizontal": 15,
          "marginTop": 10,
        }
      }
    >
      some error message
    </Text>
  </View>
</View>
`;
